rclpy/src/rclpy/_rclpy.c

```diff
@@ -2459,7 +2468,47 @@ rclpy_take(PyObject * Py_UNUSED(self), PyObject * args)
     return PyErr_NoMemory();
   }
 
-  rcl_ret_t ret = rcl_take(subscription, taken_msg, NULL);
+  rcl_ret_t ret;
+  if (raw) {
```

Comment:
shouldn't that check happen before you call `create_ros_message()` in line 2465? 
It seems like you create a message and then immediately destroy it if it's not `raw`.

Reply:
also I feel that this `if` `else` part could deserve its own function for clarity. But that's just my opinion maybe.